python/ray/_private/node.py

```diff
@@ -455,22 +461,31 @@ def _init_temp(self):
                 )
                 self._temp_dir = ray._private.utils.decode(temp_dir)
             else:
+                self._ray_params.update_if_absent(
+                    temp_dir=ray._private.utils.get_ray_temp_dir()
+                )
                 self._temp_dir = self._ray_params.temp_dir
 
+        assert self._temp_dir is not None
         try_to_create_directory(self._temp_dir)
 
         if self.head:
             self._session_dir = os.path.join(self._temp_dir, self._session_name)
         else:
-            if self._temp_dir is None or self._session_name is None:
+            if self._session_name is None:
                 assert not self._default_worker
                 session_dir = ray._private.utils.internal_kv_get_with_retry(
                     self.get_gcs_client(),
                     "session_dir",
                     ray_constants.KV_NAMESPACE_SESSION,
                     num_retries=ray_constants.NUM_REDIS_GET_RETRIES,
                 )
-                self._session_dir = ray._private.utils.decode(session_dir)
+                if sys.platform == self._head_os:
+                    self._session_dir = ray._private.utils.decode(session_dir)
+                else:
+                    self._session_dir = os.path.join(
+                        self._temp_dir, os.path.basename(session_dir)
+                    )
```

Comment:
For `session_dir`, if `session_name` is not available, it will only adopt head node's session name (basename of `session_dir`), instead of reusing the whole `session_dir`.